### ISSUE TITLE
Are repeatable Components<->Instances possible, or in the roadmap?

### ISSUE THREAD

**Issue by ijpatricio** [excalidraw/excalidraw#6918]
Resembling a Figma feature, I would like to make a "component" - imagine a naive button, with a rectangle, a text and an icon.

I want to use this "component" multiple times, and if I edit the "master", every other instance will update accordingly.
Ideally, the instances that had customizations would not change.

For example: if I change the text value on a button "instance", the text would remain. Changes to the icon or to the rectangle would be applied.

**Comment by ad1992** [comment]
HI @ijpatricio I don't think I got it. What is "master" here ? could you elaborate more or share an example in some existing app if possible ?

**Comment by ijpatricio** [comment]
Hey @ad1992 

Indeed, check the picture below.

I want to be able to make a "master/main" component (it's essentially a grouping).
I want to "clone" a component multiple times...
If I make a change in the "master/main component", then Instance 1 and Instance 2 also are affected.

Example:

- If I change "Main" to be square rectangle, then Instance 1 and 2 also become rectangle.
- If I change "Main" text from "Submit" to "Proceed", then Instance q becomes "Proceed", but Instance 2 stays "Review" because it had a change after cloning.


![image](https://github.com/excalidraw/excalidraw/assets/26031459/49480928-e8bf-4e6f-9195-1ea44578e850)

**Comment by ijpatricio** [comment]
Hello @ad1992 

This is a friendly check-in to see if I now explained better, because I love this subject, and I believe it can make the difference in using Excalidraw as a simple wireframing tool. For me and everyone 😄 

Without components, it's a pain to change dozens of the same repeated elements, after making many "screens".

**Comment by ad1992** [comment]
@ijpatricio I agree the feature would be very useful but at the same time I am scared it would not go with Excalidraw simplicity principles since in Excalidraw we want to keep things very simple and this is a feature that is not in that category.

Would like to hear what the team thinks as well
cc @dwelle
